### PR TITLE
Fix encoding of non-simple return types in RestEasy Reactive Clients

### LINKED ISSUE #21375: RestEasy Reactive Client + byte[] generates invalid classes
### Describe the bug

The following simple interface generates an invalid class file.
```kotlin
@RegisterRestClient(baseUri = "http://example.com")
interface TestAPI {

  @GET
  suspend fun getHTML(): ByteArray

}
```

Any attempt to load the class results in the following error:
```
Request failed: java.lang.ClassFormatError: Illegal class name "byte[]" in class file org/acme/TestAPI$$QuarkusRestClientInterface
```

### Expected behavior

Using Kotlin interfaces with `ByteArray`s results in valid classes.

### Actual behavior

Invalid classes are generated for the Kotlin interface with `ByteArray`s.

### How to Reproduce?

[code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/7521168/code-with-quarkus.zip)

### Output of `uname -a` or `ver`

Darwin ... 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64 x86_64

### Output of `java -version`

java version "17.0.1" 2021-10-19 LTS Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39) Java HotSpot(TM) 64-Bit Server VM (build 17.0.1+12-LTS-39, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

`2.4.1` and `HEAD` of `main`

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

mvnw 

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by kdubb**
Fixes #21375

I tracked down the error but as I'm pretty unfamiliar with the generated client testing, this still needs tests.  

The issue actually seems to affect all non-parameterized and non-primitive types. This includes all Java array types which is why `byte[]` was affected.

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

**Review by geoand — COMMENTED** [review]
Nice catch!

This is a mistake that is too easy to make unfortunately... It's happened before...

We should definitely have tests for this as well

**Comment by kdubb** [comment]
FYI, I've tested locally with both the Java `Uni<byte[]>` case and Kotlin coroutine `byte[]` cases. It fixes both of those cases.

**Comment by gsmet** [comment]
I'll you decide about the test requirement but I will push the fix to 2.4 anyway as it seems like something we want.

**Comment by geoand** [comment]
> FYI, I've tested locally with both the Java `Uni<byte[]>` case and Kotlin coroutine `byte[]` cases. It fixes both of those cases.

Excellent 👌.

Now let's just add some tests to the PR

**Review by mamadaliev — APPROVED** [review]
Looks fine, thanks.

**Comment by geoand** [comment]
Existing tests can be found at: https://github.com/quarkusio/quarkus/tree/main/extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/test/java/io/quarkus/jaxrs/client/reactive/deployment/test

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/HelloNonSimpleClient.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Path("/bytes")
+    Uni<byte[]> echoAsyncBytes(byte[] value);
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/ints")
+    Uni<Integer[]> echoAsyncInts(Integer[] value);
+
+    // @GET
```
Comment (gastaldi) [inline extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/HelloNonSimpleClient.java:42]: Delete these commented out code

Comment (kdubb) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/HelloNonSimpleClient.java:43]: 👍 Forgot to remove my sanity tests.

Comment (kdubb) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/HelloNonSimpleClient.java:43]: Removed

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by kdubb** [comment]
@geoand @gsmet It's ready to merge 😉 Just thinking this should make it into the next 2.5 and 24 releases.

**Comment by akpatternica** [comment]
@gsmet it seems you forget it for 2.5!:)

**Pull request merged**